Libs/Optimize/Optimize.cpp

```diff
@@ -76,6 +76,10 @@ bool Optimize::Run()
     return false;
   }
 
+  // ensure use_shape_statistics_after is no more than the total
+  this->m_use_shape_statistics_after = std::min(this->m_use_shape_statistics_after,
+                                                this->m_number_of_particles[0]);

```

Comment:
shouldn't it be the min of m_number_of_particles, and not m_number_of_particles[0]? (i'm assuming that this vector corresponds to multi domains)

Reply:
Yes, you're right.